platforms/core-configuration/declarative-dsl-core/src/main/kotlin/org/gradle/internal/declarativedsl/analysis/PropertyAccessResolver.kt

```diff
@@ -73,8 +78,19 @@ class PropertyAccessResolverImpl(
     @Suppress("NestedBlockDepth")
     private
     fun AnalysisContext.doResolvePropertyAccessToObject(
-        propertyAccess: PropertyAccess
+        propertyAccess: PropertyAccess,
+        expectedType: DataTypeRef?
     ): ObjectOrigin? {
+        if (propertyAccess.receiver == null && expectedType != null && expectedType is DataTypeRef.Name) {
```

Comment:
Do not assume that `expectedType is DataTypeRef.Name`. It can be a `DataTypeRef.Type` with the enum type referenced directly rather than by name.